.github/scripts/poll_batch_job_status.sh

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+# Poll an AWS Batch job to check its status.
+#
+# Takes two positional arguments:
+#
+#   1. (required) The ID of the Batch job to poll
+#   2. (optional) An enum indicating which type of poll to run: `startup` or
+#      `completion`. Defaults to `completion`. If the value of the argument is
+#      `startup`, the script will treat the `RUNNING` status as a terminal
+#      success status. Otherwise, only `SUCCESS` will count as a terminal
+#      success status, and a status of `RUNNING` will cause the script to
+#      continue polling.
+#
+# Example usage:
+#
+#   ./.github/scripts/poll_batch_job_status.sh 12345 startup
+set -euo pipefail
+
+BATCH_JOB_LOG_URL_PREFIX="https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:log-groups/log-group/%2Faws%2Fbatch%2Fjob/log-events"
+# How many times to poll AWS Batch job status while it's starting up before
+# deciding to raise an error. Multiply by BATCH_JOB_POLL_INTERVAL_SECONDS to
+# derive a timeout in second units. There is no equivalent timeout for running
+# jobs, because those timeouts can be set on the Batch level, whereas startup
+# timeouts are not controllable by Batch
+BATCH_JOB_POLL_STARTUP_MAX_RETRIES=60
```

Comment:
It could make sense to eventually make these attributes configurable, particularly once we factor this logic out into a shared action or workflow, but for the sake of faster iteration I decided to hardcode them for now since bash argument parsing is so annoying.